src/main.rs

```diff
@@ -8,13 +8,14 @@ fn app_logic(data: &mut i32) -> impl View<i32> {
     } else {
         format!("clicked {data} times")
     };
+    let widget: Box<dyn AnyView<i32, WidgetBound> + Send> = Box::new(button(label, |data| {
+        println!("clicked");
+        *data += 1;
+    }));
 
     // The actual UI Code starts here
     v_stack((
-        button(label, |data| {
-            println!("clicked");
-            *data += 1;
-        }),
+        widget,
```

Comment:
This seems like a bit of a regression in ergonomics:

- Will end users be expected to write out types like this?
- Will children always be required to be boxed like this?